packages/keyring-controller/src/KeyringController.ts

```diff
@@ -279,11 +279,11 @@ export class KeyringController extends BaseController<
    * @returns Newly-created keychain object.
    */
   async createNewVaultAndKeychain(password: string) {
-    const releaseLock = await this.mutex.acquire();
+    const releaseLock = await this.#mutex.acquire();
     try {
       const vault = await this.#keyring.createNewVaultAndKeychain(password);
-      this.updateIdentities(await this.#keyring.getAccounts());
-      this.fullUpdate();
+      this.#updateIdentities(await this.#keyring.getAccounts());
+      await this.fullUpdate();
```

Comment:
This potentially changes the behaviour of this function. We may want to do this instead:

```suggestion
      this.fullUpdate().catch(console.error);
```

Reply:
Makes sense to me.